### ISSUE TITLE
Issues with Samsung Galaxy devices (NullPointerException in MusicBinder, MusicModule IllegaStatelException)

### ISSUE THREAD

**Issue by DennisdeWitNL** [DoubleSymmetry/react-native-track-player#1210]
**Describe the bug**

Dear,

I can't very much describe the bug, I just see a lot of crashes happening with mostly Samsung Galaxy (around 70% of the devices are Samsung Galaxy) and some Xiaomi's. I noticed two crashes that are very common in my app.

Mainly:

```
java.lang.NullPointerException: 
  at com.guichaguri.trackplayer.service.MusicBinder.post (MusicBinder.java:22)
  at com.guichaguri.trackplayer.module.MusicModule.waitForConnection (MusicModule.java:94)
  at com.guichaguri.trackplayer.module.MusicModule.stop (MusicModule.java:334)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.facebook.react.bridge.JavaMethodWrapper.invoke (JavaMethodWrapper.java:372)
  at com.facebook.react.bridge.JavaModuleWrapper.invoke (JavaModuleWrapper.java:151)
  at com.facebook.react.bridge.queue.NativeRunnable.run (Native Method)
  at android.os.Handler.handleCallback (Handler.java:938)
  at android.os.Handler.dispatchMessage (Handler.java:99)
  at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage (MessageQueueThreadHandler.java:27)
  at android.os.Looper.loop (Looper.java:246)
  at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run (MessageQueueThreadImpl.java:226)
  at java.lang.Thread.run (Thread.java:923)
```

This is the most common crash. The other one that is most common is:

```
java.lang.IllegalStateException: 
  at android.app.ContextImpl.startServiceCommon (ContextImpl.java:1795)
  at android.app.ContextImpl.startService (ContextImpl.java:1740)
  at android.content.ContextWrapper.startService (ContextWrapper.java:738)
  at android.content.ContextWrapper.startService (ContextWrapper.java:738)
  at com.guichaguri.trackplayer.module.MusicModule.waitForConnection (MusicModule.java:106)
  at com.guichaguri.trackplayer.module.MusicModule.add (MusicModule.java:195)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.facebook.react.bridge.JavaMethodWrapper.invoke (JavaMethodWrapper.java:372)
  at com.facebook.react.bridge.JavaModuleWrapper.invoke (JavaModuleWrapper.java:151)
  at com.facebook.react.bridge.queue.NativeRunnable.run (Native Method)
  at android.os.Handler.handleCallback (Handler.java:938)
  at android.os.Handler.dispatchMessage (Handler.java:99)
  at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage (MessageQueueThreadHandler.java:27)
  at android.os.Looper.loop (Looper.java:246)
  at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run (MessageQueueThreadImpl.java:226)
  at java.lang.Thread.run (Thread.java:923)
```

**To Reproduce**
I have no clue how to reproduce it. I never experienced the problem myself.

**Environment (please complete the following information):**
Run `react-native info` in your project and share the content.

What `react-native-track-player` version are you using?
Version 1.2.7

Are you testing on a real device or in the simulator? Which OS version are you running?
Real devices. I see these crashes in Play Console, mostly reported on Android 11 and Android 10.

**Code**
In Home.js (mainscreen)

```
const setup = async () => {
  await TrackPlayer.setupPlayer({});
  await TrackPlayer.updateOptions({
    stopWithApp: true,
    capabilities: [
      TrackPlayer.CAPABILITY_PLAY,
      TrackPlayer.CAPABILITY_PAUSE,
      TrackPlayer.CAPABILITY_STOP
    ],
    compactCapabilities: [ 
      TrackPlayer.CAPABILITY_PLAY,
      TrackPlayer.CAPABILITY_PAUSE,
      TrackPlayer.CAPABILITY_STOP
    ],
  })};
setup();
```

**The Player.js (player)**
 ```
async InitTrackPlayer() {


   this.setState({paused: false});
       const { navigation } = this.props;
    const streamID = navigation.getParam('streamID', 'NO-ID');


      fetch(`https://xxx.xxx.xxx/getinfoviajson.php?streamID=${streamID}`, {
          method: "POST",
    })
      .then(response => response.json())
      .then(responseJson => {
        this.setState({
          streamName: responseJson.streamName,
          desc: responseJson.streamDesc_app,
          mountpoint: responseJson.mountpoint,
          fullStreamURL: responseJson.fullStreamURL,
          imageURL: responseJson.imageURL,
          phone_app: responseJson.phone_app,
          phone_app_plus: responseJson.phone_app_plus,
          streamURLJSON: responseJson.streamURL,
        });


var streambyJSON = this.state.streamURLJSON

const start = async () => {

  // Add a track to the queue
  await TrackPlayer.add({
    id: `${streamID}`,
    url: streambyJSON,
    artist: '',
    title: '',
    artwork: '',
});

};

start();





});


      };

```

The snippet to update the metadata

```
 async trackupdate() {
        const trackId = await TrackPlayer.getCurrentTrack();
    await TrackPlayer.updateMetadataForTrack(trackId, {
        title: this.state.nowPlaying,
        artist: this.state.streamName,
        artwork: this.state.imageURL,
    });
}
```

**The snippet that actions to certain statusses (set the correct button or reset the stream when the new stream mountpoint is not the same as the stream of a previous mountpoint)**

```
async getPlayerStatus() {
 const { navigation } = this.props;
 const streamID = navigation.getParam('streamID', 'NO-ID');  
let state = await TrackPlayer.getState();
let thisIsPlaying = await TrackPlayer.getCurrentTrack();

if (state == 'playing') {
  this.setState({DoesPlay: true});
}

if (state == 'idle') {
  this.setState({DoesPlay: false});
}

if (state == 'ready') {
  this.setState({DoesPlay: false});
}

if (state == '1') {
  this.setState({DoesPlay: false});
}

if (state == '2') {
  this.setState({DoesPlay: false});
}

if (state == '3') {
  this.setState({DoesPlay: true});
}

if (thisIsPlaying != streamID)  {
  var statePlayer = this.state.DoesPlay;
  this.setState({DoesPlay: false});
   // alert(statePlayer);
   //  alert (state);
   if (Platform.OS === 'ios') {
    await TrackPlayer.stop();
  }
else {
  const { navigation } = this.props;
  const streamID = navigation.getParam('streamID', 'NO-ID');
     await TrackPlayer.reset();
  }


}
}
```

**componentDidMount and componentWillUnmount**

```
 componentDidMount() {
this.getPlayerStatus();

   this.setState({paused: false});
       const { navigation } = this.props;


 this.InitTrackPlayer();
 this.trackupdate();


        TrackPlayer.addEventListener('remote-play', () => {
TrackPlayer.play();
  });

    TrackPlayer.addEventListener('remote-pause', () => {
this.PauseClick();
  });
 
    TrackPlayer.addEventListener('remote-duck', () => {
  TrackPlayer.stop();
  this.setState({DoesPlay: false});
  this.InitTrackPlayer();

  }); 

    TrackPlayer.addEventListener('remote-stop', () => {
    TrackPlayer.stop();
  }); 

}

componentWillUnmount() {

}
```

Any idea's what causes this problem? Thanks guys!

**Comment by dcvz** [comment]
@DennisdeWitNL does this happen on the v2 branch as well?

**Comment by phillipcmittmann** [comment]
The first crash i think it resolved on 2.0.1. Applied the same solution to a fork.

Can confirm second crash happening in v2.0.0-rc17, on androids 8+ in some devices, being more common in Samsungs and Motorolas.

I'm no android dev, but i think the solution can be found here [https://stackoverflow.com/a/59810618](url)

**Comment by DennisdeWitNL** [comment]
@dcvz Yes. I still see the MusicBinder error with version @2.0.1, however, much less than before. It only occured 8 times today.

**Comment by supernaut** [comment]
I've migrated an app from 1.x to 2.0.3 and this happens every time the app starts on Andriod.

```
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.radio.me10a03245, PID: 6627
    java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.Integer.intValue()' on a null object reference
        at com.guichaguri.trackplayer.service.metadata.MetadataManager.updateOptions(MetadataManager.java:114)
        at com.guichaguri.trackplayer.service.MusicBinder.updateOptions(MusicBinder.java:49)
        at com.guichaguri.trackplayer.module.MusicModule.onServiceConnected(MusicModule.java:76)
        at android.app.LoadedApk$ServiceDispatcher.doConnected(LoadedApk.java:1954)
        at android.app.LoadedApk$ServiceDispatcher$RunConnection.run(LoadedApk.java:1986)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:223)
        at android.app.ActivityThread.main(ActivityThread.java:7656)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
```

**Comment by dcvz** [comment]
@supernaut your particular issue is likely due to calling some methods with the wrong parameter types.

There was breaking changes in v2 - particularly about how the queue is managed. The internals are based on indexes now instead of track ID's. 

It will help if you refer to the migration part of the README.

**Comment by supernaut** [comment]
I have read the documentation and made all documented changes to my code. It builds as expected and runs as expected on iOS. When starting the app on Android it crashes on startup with the above error.

**Comment by austinried** [comment]
Can confirm that I'm seeing the second crash as well, I think it's the same or similar stack trace:

```
java.lang.IllegalStateException: 
  at android.app.ContextImpl.startServiceCommon (ContextImpl.java:1795)
  at android.app.ContextImpl.startService (ContextImpl.java:1740)
  at android.content.ContextWrapper.startService (ContextWrapper.java:738)
  at android.content.ContextWrapper.startService (ContextWrapper.java:738)
  at com.guichaguri.trackplayer.module.MusicModule.waitForConnection (MusicModule.java:29)
  at com.guichaguri.trackplayer.module.MusicModule.getQueue (MusicModule.java:5)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.facebook.react.bridge.JavaMethodWrapper.invoke (Unknown Source:147)
  at com.facebook.react.bridge.JavaModuleWrapper.invoke (Unknown Source:21)
  at com.facebook.react.bridge.queue.NativeRunnable.run (Native Method)
  at android.os.Handler.handleCallback (Handler.java:938)
  at android.os.Handler.dispatchMessage (Handler.java:99)
  at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage (Unknown Source)
  at android.os.Looper.loop (Looper.java:246)
  at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run (Unknown Source:37)
  at java.lang.Thread.run (Thread.java:923)
```

Appears to be happening on Samsung Galaxy S21 5G and OnePlus OnePlus6T phones, Android 10 and 11.  My app is running react-native-track-player 2.0.3.

This is my app if seeing any code helps: https://github.com/austinried/subtracks
Playing anything primarily happens here: https://github.com/austinried/subtracks/blob/master/app/state/trackplayer.ts#L201

**Comment by redepampa** [comment]
> **Describe the bug**
> 
> Dear,
> 
> I can't very much describe the bug, I just see a lot of crashes happening with mostly Samsung Galaxy (around 70% of the devices are Samsung Galaxy) and some Xiaomi's. I noticed two crashes that are very common in my app.
> 
> Mainly:
> 
> ```
> java.lang.NullPointerException: 
>   at com.guichaguri.trackplayer.service.MusicBinder.post (MusicBinder.java:22)
>   at com.guichaguri.trackplayer.module.MusicModule.waitForConnection (MusicModule.java:94)
>   at com.guichaguri.trackplayer.module.MusicModule.stop (MusicModule.java:334)
>   at java.lang.reflect.Method.invoke (Native Method)
>   at com.facebook.react.bridge.JavaMethodWrapper.invoke (JavaMethodWrapper.java:372)
>   at com.facebook.react.bridge.JavaModuleWrapper.invoke (JavaModuleWrapper.java:151)
>   at com.facebook.react.bridge.queue.NativeRunnable.run (Native Method)
>   at android.os.Handler.handleCallback (Handler.java:938)
>   at android.os.Handler.dispatchMessage (Handler.java:99)
>   at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage (MessageQueueThreadHandler.java:27)
>   at android.os.Looper.loop (Looper.java:246)
>   at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run (MessageQueueThreadImpl.java:226)
>   at java.lang.Thread.run (Thread.java:923)
> ```
> 
> This is the most common crash. The other one that is most common is:
> 
> ```
> java.lang.IllegalStateException: 
>   at android.app.ContextImpl.startServiceCommon (ContextImpl.java:1795)
>   at android.app.ContextImpl.startService (ContextImpl.java:1740)
>   at android.content.ContextWrapper.startService (ContextWrapper.java:738)
>   at android.content.ContextWrapper.startService (ContextWrapper.java:738)
>   at com.guichaguri.trackplayer.module.MusicModule.waitForConnection (MusicModule.java:106)
>   at com.guichaguri.trackplayer.module.MusicModule.add (MusicModule.java:195)
>   at java.lang.reflect.Method.invoke (Native Method)
>   at com.facebook.react.bridge.JavaMethodWrapper.invoke (JavaMethodWrapper.java:372)
>   at com.facebook.react.bridge.JavaModuleWrapper.invoke (JavaModuleWrapper.java:151)
>   at com.facebook.react.bridge.queue.NativeRunnable.run (Native Method)
>   at android.os.Handler.handleCallback (Handler.java:938)
>   at android.os.Handler.dispatchMessage (Handler.java:99)
>   at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage (MessageQueueThreadHandler.java:27)
>   at android.os.Looper.loop (Looper.java:246)
>   at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run (MessageQueueThreadImpl.java:226)
>   at java.lang.Thread.run (Thread.java:923)
> ```
> 
> **To Reproduce**
> I have no clue how to reproduce it. I never experienced the problem myself.
> 
> **Environment (please complete the following information):**
> Run `react-native info` in your project and share the content.
> 
> What `react-native-track-player` version are you using?
> Version 1.2.7
> 
> Are you testing on a real device or in the simulator? Which OS version are you running?
> Real devices. I see these crashes in Play Console, mostly reported on Android 11 and Android 10.
> 
> **Code**
> In Home.js (mainscreen)
> 
> ```
> const setup = async () => {
>   await TrackPlayer.setupPlayer({});
>   await TrackPlayer.updateOptions({
>     stopWithApp: true,
>     capabilities: [
>       TrackPlayer.CAPABILITY_PLAY,
>       TrackPlayer.CAPABILITY_PAUSE,
>       TrackPlayer.CAPABILITY_STOP
>     ],
>     compactCapabilities: [ 
>       TrackPlayer.CAPABILITY_PLAY,
>       TrackPlayer.CAPABILITY_PAUSE,
>       TrackPlayer.CAPABILITY_STOP
>     ],
>   })};
> setup();
> ```
> 
> **The Player.js (player)**
> 
> ```
> async InitTrackPlayer() {
> 
> 
>   this.setState({paused: false});
>       const { navigation } = this.props;
>    const streamID = navigation.getParam('streamID', 'NO-ID');
> 
> 
>      fetch(`https://xxx.xxx.xxx/getinfoviajson.php?streamID=${streamID}`, {
>          method: "POST",
>    })
>      .then(response => response.json())
>      .then(responseJson => {
>        this.setState({
>          streamName: responseJson.streamName,
>          desc: responseJson.streamDesc_app,
>          mountpoint: responseJson.mountpoint,
>          fullStreamURL: responseJson.fullStreamURL,
>          imageURL: responseJson.imageURL,
>          phone_app: responseJson.phone_app,
>          phone_app_plus: responseJson.phone_app_plus,
>          streamURLJSON: responseJson.streamURL,
>        });
> 
> 
> var streambyJSON = this.state.streamURLJSON
> 
> const start = async () => {
> 
>  // Add a track to the queue
>  await TrackPlayer.add({
>    id: `${streamID}`,
>    url: streambyJSON,
>    artist: '',
>    title: '',
>    artwork: '',
> });
> 
> };
> 
> start();
> 
> 
> 
> 
> 
> });
> 
> 
>      };
> ```
> 
> The snippet to update the metadata
> 
> ```
>  async trackupdate() {
>         const trackId = await TrackPlayer.getCurrentTrack();
>     await TrackPlayer.updateMetadataForTrack(trackId, {
>         title: this.state.nowPlaying,
>         artist: this.state.streamName,
>         artwork: this.state.imageURL,
>     });
> }
> ```
> 
> **The snippet that actions to certain statusses (set the correct button or reset the stream when the new stream mountpoint is not the same as the stream of a previous mountpoint)**
> 
> ```
> async getPlayerStatus() {
>  const { navigation } = this.props;
>  const streamID = navigation.getParam('streamID', 'NO-ID');  
> let state = await TrackPlayer.getState();
> let thisIsPlaying = await TrackPlayer.getCurrentTrack();
> 
> if (state == 'playing') {
>   this.setState({DoesPlay: true});
> }
> 
> if (state == 'idle') {
>   this.setState({DoesPlay: false});
> }
> 
> if (state == 'ready') {
>   this.setState({DoesPlay: false});
> }
> 
> if (state == '1') {
>   this.setState({DoesPlay: false});
> }
> 
> if (state == '2') {
>   this.setState({DoesPlay: false});
> }
> 
> if (state == '3') {
>   this.setState({DoesPlay: true});
> }
> 
> if (thisIsPlaying != streamID)  {
>   var statePlayer = this.state.DoesPlay;
>   this.setState({DoesPlay: false});
>    // alert(statePlayer);
>    //  alert (state);
>    if (Platform.OS === 'ios') {
>     await TrackPlayer.stop();
>   }
> else {
>   const { navigation } = this.props;
>   const streamID = navigation.getParam('streamID', 'NO-ID');
>      await TrackPlayer.reset();
>   }
> 
> 
> }
> }
> ```
> 
> **componentDidMount and componentWillUnmount**
> 
> ```
>  componentDidMount() {
> this.getPlayerStatus();
> 
>    this.setState({paused: false});
>        const { navigation } = this.props;
> 
> 
>  this.InitTrackPlayer();
>  this.trackupdate();
> 
> 
>         TrackPlayer.addEventListener('remote-play', () => {
> TrackPlayer.play();
>   });
> 
>     TrackPlayer.addEventListener('remote-pause', () => {
> this.PauseClick();
>   });
>  
>     TrackPlayer.addEventListener('remote-duck', () => {
>   TrackPlayer.stop();
>   this.setState({DoesPlay: false});
>   this.InitTrackPlayer();
> 
>   }); 
> 
>     TrackPlayer.addEventListener('remote-stop', () => {
>     TrackPlayer.stop();
>   }); 
> 
> }
> 
> componentWillUnmount() {
> 
> }
> ```
> 
> Any idea's what causes this problem? Thanks guys!

any update on this?

**Comment by bourgois** [comment]
@austinried @supernaut @DennisdeWitNL @phillipcmittmann any of you found a solution? We have updated from V1 to V2.1 and we are going from 0.x% crashes to more than 1% now that it's been deployed for a few days (we have around 10k users on this version).

**Comment by phillipcmittmann** [comment]
> @austinried @supernaut @DennisdeWitNL @phillipcmittmann any of you found a solution? We have updated from V1 to V2.1 and we are going from 0.x% crashes to more than 1% now that it's been deployed for a few days (we have around 10k users on this version).

To be honest, haven't touched this since that time because another priority surged. But happened a lot more since last time.

I tried to debug when we talked but couldn't find anything.

**Comment by dcvz** [comment]
We're rewriting our Android module and making the binder/service/notification parts more stable and robust as part of https://github.com/DoubleSymmetry/react-native-track-player/discussions/1264


We expect this issue to be fixed as part of that effort. We're aiming to have the first pre-release soon.

**Comment by bourgois** [comment]
> We're rewriting our Android module and making the binder/service/notification parts more stable and robust as part of #1264
> 
> We expect this issue to be fixed as part of that effort. We're aiming to have the first pre-release soon.

Do you have a rough timing expectation? We can't reasonably revert to the V1, but if the Kotlin version isn't going to be available soon we need to work on a solution for V2 .

**Comment by peterchijioke** [comment]
how do I resolve the above (Android)
, I am using version 2.0.1

**Comment by dcvz** [comment]
> > We're rewriting our Android module and making the binder/service/notification parts more stable and robust as part of #1264
> > We expect this issue to be fixed as part of that effort. We're aiming to have the first pre-release soon.
> 
> Do you have a rough timing expectation? We can't reasonably revert to the V1, but if the Kotlin version isn't going to be available soon we need to work on a solution for V2 .

We hope to have the first pre-release this week.

**Comment by bourgois** [comment]
> > > We're rewriting our Android module and making the binder/service/notification parts more stable and robust as part of #1264
> > > We expect this issue to be fixed as part of that effort. We're aiming to have the first pre-release soon.
> > 
> > 
> > Do you have a rough timing expectation? We can't reasonably revert to the V1, but if the Kotlin version isn't going to be available soon we need to work on a solution for V2 .
> 
> We hope to have the first pre-release this week.

Great! Don't hesitate to ping us so we can test it on our side and give you feedbacks (probably not on those crashs as we see them only in the wild)

**Comment by xaiamov** [comment]
> java.lang.NullPointerException: 
>   at com.guichaguri.trackplayer.service.MusicBinder.post (MusicBinder.java:22)
>   at com.guichaguri.trackplayer.module.MusicModule.waitForConnection (MusicModule.java:94)
>   at com.guichaguri.trackplayer.module.MusicModule.stop (MusicModule.java:334)

```
    public void post(Runnable r) {
        service.handler.post(r);
    }
```
This seem to happen when music service doesn't exist during TrackPlayer.stop() command. 
Maybe it appears when a user kills the app manually?

**Comment by MANTENN** [comment]
> I've migrated an app from 1.x to 2.0.3 and this happens every time the app starts on Andriod.
> 
> ```
> E/AndroidRuntime: FATAL EXCEPTION: main
>     Process: com.APPNAME, PID: 6627
>     java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.Integer.intValue()' on a null object reference
>         at com.guichaguri.trackplayer.service.metadata.MetadataManager.updateOptions(MetadataManager.java:114)
>         at com.guichaguri.trackplayer.service.MusicBinder.updateOptions(MusicBinder.java:49)
>         at com.guichaguri.trackplayer.module.MusicModule.onServiceConnected(MusicModule.java:76)
>         at android.app.LoadedApk$ServiceDispatcher.doConnected(LoadedApk.java:1954)
>         at android.app.LoadedApk$ServiceDispatcher$RunConnection.run(LoadedApk.java:1986)
>         at android.os.Handler.handleCallback(Handler.java:938)
>         at android.os.Handler.dispatchMessage(Handler.java:99)
>         at android.os.Looper.loop(Looper.java:223)
>         at android.app.ActivityThread.main(ActivityThread.java:7656)
>         at java.lang.reflect.Method.invoke(Native Method)
>         at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
>         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
> ```

This is what I get:

```
FATAL EXCEPTION: main

java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.Integer.intValue()' on a null object reference

FATAL EXCEPTION: main
Process: com.x, PID: 10825
java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.Integer.intValue()' on a null object reference
	at com.guichaguri.trackplayer.service.e.b.q(Unknown Source:42)
	at com.guichaguri.trackplayer.service.b.h(Unknown Source:29)
	at com.guichaguri.trackplayer.module.MusicModule.lambda$updateOptions$1(Unknown Source:4)
	at com.guichaguri.trackplayer.module.MusicModule.C(Unknown Source:0)
	at com.guichaguri.trackplayer.module.r.run(Unknown Source:4)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:246)
	at android.app.ActivityThread.main(ActivityThread.java:8633)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)
```

My code:
```
TrackPlayer.setupPlayer().then(async (r) => {
                      await TrackPlayer.updateOptions({
                        // One of RATING_HEART, RATING_THUMBS_UP_DOWN, RATING_3_STARS, RATING_4_STARS, RATING_5_STARS, RATING_PERCENTAGE
                        ratingType: TrackPlayer.RATING_5_STARS,

                        // Whether the player should stop running when the app is closed on Android
                        stopWithApp: true,
                        alwaysPauseOnInterruption: true,
                        capabilities: [
                          TrackPlayer.CAPABILITY_PLAY,
                          TrackPlayer.CAPABILITY_PAUSE,
                          TrackPlayer.CAPABILITY_STOP,
                          TrackPlayer.CAPABILITY_SKIP_TO_NEXT,
                          TrackPlayer.CAPABILITY_SKIP_TO_PREVIOUS,
                          TrackPlayer.CAPABILITY_SEEK_TO,
                        ],

                        // An array of capabilities that will show up when the notification is in the compact form on Android
                        compactCapabilities: [
                          TrackPlayer.CAPABILITY_PLAY,
                          TrackPlayer.CAPABILITY_PAUSE,
                          TrackPlayer.CAPABILITY_STOP,
                          TrackPlayer.CAPABILITY_SKIP_TO_NEXT,
                          TrackPlayer.CAPABILITY_SKIP_TO_PREVIOUS,
                          TrackPlayer.CAPABILITY_SEEK_TO,
                        ],
                        notificationCapabilities: [
                          TrackPlayer.CAPABILITY_PLAY,
                          TrackPlayer.CAPABILITY_PAUSE,
                          TrackPlayer.CAPABILITY_STOP,
                          TrackPlayer.CAPABILITY_SKIP_TO_NEXT,
                          TrackPlayer.CAPABILITY_SKIP_TO_PREVIOUS,
                          TrackPlayer.CAPABILITY_SEEK_TO,
                        ],
                        // icon: require('../img/icon.png'),
                      })
                        .then(async (r) => {
                          const localFile = await checkLocalTrackFirst(
                            {id: i, url: urlHandler(file.url)},
                            albumDir,
                          );
                          console.log('localFile', localFile);
                          return {r, localFile};
                        })
                        .then(async (r) => {
                          // Add a track to the queue
                          // await TrackPlayer.removeUpcomingTracks();
                          console.log('fileUrl', urlHandler(file.url));
                          console.log('r', r.localFile);
                          await TrackPlayer.stop();
                          return await TrackPlayer.add({
                            id: i,
                            url: r.localFile
                              ? r.localFile
                              : urlHandler(file.url),
                            title: name,
                            artist: data.albumsConnection.albums[0].group.name,
                            artwork: urlHandler(
                              data.albumsConnection.albums[0].artwork[0].url,
                            ),
                            duration: duration / 1000,
                          }).then((r) => {
                            TrackPlayer.play();
                          });
                        });
                    }); // Start playing it
                    ```

**Comment by MANTENN** [comment]
I think this might be related: http://react-native-track-player.js.org/getting-started/ Capabilities have changed; 

https://github.com/DoubleSymmetry/react-native-track-player/issues/665

**Comment by MANTENN** [comment]
It happens because of `capabilities` option the other ones I passed it works. Not sure how that implementation differs vs the others. In the mean time I am using react native music controls

**Comment by bradleyflood** [comment]
Expected to be resolved in #1371 

Thanks all
